src/packages/admin-ui-components/src/detail-panel/fields/markdown-field/menu-bar.tsx

```diff
@@ -0,0 +1,408 @@
+		event.preventDefault();
+		event.stopPropagation();
+		editor.chain().focus().setParagraph().run();
+	};
+
+	return (
+		<div onMouseEnter={() => setShowItems(true)} onMouseLeave={() => setShowItems(false)}>
+			{show.h1 ? (
+				<button
+					onClick={handleHeadingClick(1)}
+					className={
+						editor.isActive('paragraph') || editor.isActive('heading') ? styles.isActive : ''
+					}
+				>
+					<EditorHIcon />
+				</button>
+			) : (
+				<button
+					onClick={handleParagraphClick}
+					className={editor.isActive('paragraph') ? styles.isActive : ''}
+				>
+					<EditorPIcon />
+				</button>
+			)}
+			{showItems && (
+				<div className={styles.itemsContainer}>
+					<button
+						onClick={handleParagraphClick}
+						className={editor.isActive('paragraph') ? styles.isActive : ''}
+					>
+						<EditorPIcon />
+					</button>
+					{show.h1 && (
+						<button
+							onClick={handleHeadingClick(1)}
+							className={editor.isActive('heading', { level: 1 }) ? styles.isActive : ''}
+						>
+							<EditorH1Icon />
+						</button>
+					)}
+					{show.h2 && (
+						<button
+							onClick={handleHeadingClick(2)}
+							className={editor.isActive('heading', { level: 2 }) ? styles.isActive : ''}
+						>
+							<EditorH2Icon />
+						</button>
+					)}
+					{show.h3 && (
+						<button
+							onClick={handleHeadingClick(3)}
+							className={editor.isActive('heading', { level: 3 }) ? styles.isActive : ''}
+						>
+							<EditorH3Icon />
+						</button>
+					)}
+					{show.h4 && (
+						<button
+							onClick={handleHeadingClick(4)}
+							className={editor.isActive('heading', { level: 4 }) ? styles.isActive : ''}
+						>
+							<EditorH4Icon />
+						</button>
+					)}
+					{show.h5 && (
+						<button
+							onClick={handleHeadingClick(5)}
+							className={editor.isActive('heading', { level: 5 }) ? styles.isActive : ''}
+						>
+							<EditorH5Icon />
+						</button>
+					)}
+					{show.h6 && (
+						<button
+							onClick={handleHeadingClick(6)}
+							className={editor.isActive('heading', { level: 6 }) ? styles.isActive : ''}
+						>
+							<EditorH6Icon />
+						</button>
+					)}
+				</div>
+			)}
+		</div>
+	);
+};
+
+const ListOptions = (props: SectionProps) => {
+	const { editor, options } = props;
+	const [showItems, setShowItems] = useState(false);
+
+	const handleBulletListClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+		event.preventDefault();
+		event.stopPropagation();
+		editor.chain().focus().toggleBulletList().run();
+	};
+
+	const handleOrderedListClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+		event.preventDefault();
+		event.stopPropagation();
+		editor.chain().focus().toggleOrderedList().run();
+	};
+
+	const handleBlockquoteClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+		event.preventDefault();
+		event.stopPropagation();
+		editor.chain().focus().toggleBlockquote().run();
+	};
+
+	const getMainAndOptionButtons = () => {
```

Comment:
This should be inside a useCallback, otherwise the elements are fresh on every render